controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/RemoveHostTask.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright 2017 Pinterest, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.pinterest.rocksplicator.controller.tasks;
+
+import com.pinterest.rocksdb_admin.thrift.Admin;
+import com.pinterest.rocksplicator.controller.bean.ClusterBean;
+import com.pinterest.rocksplicator.controller.bean.HostBean;
+import com.pinterest.rocksplicator.controller.bean.SegmentBean;
+import com.pinterest.rocksplicator.controller.config.ConfigParser;
+import com.pinterest.rocksplicator.controller.util.AdminClientFactory;
+import com.pinterest.rocksplicator.controller.util.ZKUtil;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+/**
+ * This task removes a given host from a cluster. It consists of the following steps:
+ *
+ * 1) Check if the host is still up and running. Fail if yes.
+ * 2) Update cluster config by removing that host.
+ *
+ * @author Ang Xu (angxu@pinterest.com)
+ */
+public class RemoveHostTask extends TaskBase<RemoveHostTask.Param> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HealthCheckTask.class);
+
+  @Inject
+  private CuratorFramework zkClient;
+
+  @Inject
+  private AdminClientFactory clientFactory;
+
+  public RemoveHostTask(HostBean hostToRemove) {
+    this(new Param().setHostToRemove(hostToRemove));
+  }
+
+  public RemoveHostTask(Param param) {
+    super(param);
+  }
+
+  @Override
+  public void process(Context ctx) throws Exception {
+    final String clusterName = ctx.getCluster();
+    final HostBean toRemove = getParameter().getHostToRemove();
+
+    InetSocketAddress hostAddr = new InetSocketAddress(toRemove.getIp(), toRemove.getPort());
+    Admin.Client client = clientFactory.getClient(hostAddr);
+
+    // 1) ping the host to remove to make sure it's not running
```

Comment:
maybe this check is too strict. sometimes we may want to remove a running but having other issue's host.